Title: Journal club: schedule for Spring 2019
Date: 2019-01-27
Slug: journal-club-spring-2019
Tags: bioinformatics, cybersecurity
Category: Journal Club
Author: Humberto Ortiz-Zuazaga

# Spring 2019

We're meeting Fridays at 9:00 AM in the lab. Here's the schedule of
journal club presentations for this semester.

| Date | Person | Paper |
|------|--------|-------|
| Feb 8 | Angel Sanquiche | Axolotl |
| Feb 15 | | *cancelled* |
| Feb 22 | Louis Gil / Roberto Lopez | Oshlack / Streams |
| Mar 1 | Israel Dilan | Next-gen Sequence Assembly |
| Mar 8 | Walter Baez | kalisto / Yanagi |
| Mar 15 | Briknie / Omar | Kmers / Kevlar |
| Mar 22 | | **Holiday (abolition of slavery)** |
| Mar 29 | Humberto | OMH  |
| Apr 5  |  Miguel | mapping |
| Apr 12 | | **Postponed: UCSC Genome browser** |
| Apr 19 | | **Spring break** |
| Apr 26 | Heriberto | subspace |

# Papers

Yanagi
: Fast and interpretable alternative splicing and differential
gene-level expression analysis using transcriptome segmentation with
Yanagi.  Mohamed K Gunady, Steve Mount, Hector Corrada Bravo.  bioRxiv
364281; doi: <https://doi.org/10.1101/364281>

Axolotl
: Stewart R, Rascón CA, Tian S, Nie J, Barry C, et al. (2013).
Comparative RNA-seq Analysis in the Unsequenced Axolotl: The Oncogene
Burst Highlights Early Gene Expression in the Blastema. PLOS
Computational Biology 9(3):
e1002936. <https://doi.org/10.1371/journal.pcbi.1002936>

Oshlack
: Fast and accurate differential transcript usage by testing
equivalence class counts.  Marek Cmero, Nadia Davidson, Alicia
Oshlack.  bioRxiv 501106; doi: <https://doi.org/10.1101/501106>

Streams
: Zhang Q, Awad S, Brown CT. 2015. Crossing the streams: a framework
for streaming analysis of short DNA sequencing reads. PeerJ PrePrints
3:e890v1 <https://doi.org/10.7287/peerj.preprints.890v1>

Next-Generation Sequence Assembly
: El-Metwally S, Hamza T, Zakaria M, Helmy M (2013) Next-Generation Sequence Assembly: Four Stages of Data Processing and Computational Challenges. PLoS Comput Biol 9(12): e1003345. <https://doi.org/10.1371/journal.pcbi.1003345>

Kevlar
:  Kevlar: a mapping-free framework for accurate discovery of de novo variants
Daniel S Standage, C Titus Brown, Fereydoun Hormozdiari
bioRxiv 549154; doi: <https://doi.org/10.1101/549154>

kmers
: Zhang, Q., Pell, J., Canino-Koning, R., Howe, A. C., & Brown,
C. T. (2014). These are not the k-mers you are looking for: efficient
online k-mer counting using a probabilistic data structure. PloS one,
9(7), e101271. <https://doi.org/10.1371/journal.pone.0101271>

mapping
: Rahman, A., Hallgrímsdóttir, I., Eisen, M., & Pachter,
L. (2018). Association mapping from sequencing reads using
k-mers. eLife, 7, e32920 <https://dx.doi.org/10.7554%2FeLife.32920>

order min hash
:  Locality sensitive hashing for the edit distance
Guillaume Marcais, Dan DeBlasio, Prashant Pandey, Carl Kingsford
bioRxiv 534446; doi: <https://doi.org/10.1101/534446> 

subspace
: Lakhina, Anukool, Mark Crovella, and Christiphe
Diot. "Characterization of network-wide anomalies in traffic flows."
Proceedings of the 4th ACM SIGCOMM conference on Internet
measurement. ACM, 2004. <https://doi.org/10.1145/1028788.1028813>
